src/dev/build/tasks/generate_packages_optimized_assets.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { pipeline } from 'stream';
+import { promisify } from 'util';
+
+import gulpBrotli from 'gulp-brotli';
```

Comment:
This is at odds with https://github.com/elastic/kibana/issues/85694 - thoughts @tylersmalley?

Reply:
My thoughts are that using gulp-* packages to accomplish generic build tasks over lots of objects outside of the gulp runner are fine, they're very useful packages and work well. Having a lot of task runners is more the issue we're trying to solve with https://github.com/elastic/kibana/issues/85694 I think.

Reply:
I agree with what @spalger wrote here, we are not actually run gulp here as a task runner 😃

Reply:
I am fine with that. I don't really care for these packages, but I understand they provide a consistent and easier to use interface for this case.